tests/output-parser_spec.lua

```diff
@@ -1,9 +1,44 @@
+local output_parser = require("neotest-kotlin.src.output-parser")
+local nio = require("nio")
+
 describe("output-parser", function()
-	local output_parser = require("neotest-kotlin.src.output-parser")
+	local example_project_path = vim.fs.joinpath(
+		debug.getinfo(1).source:match("@?(.*/)"),
+		"example_project",
+		"app",
+		"src",
+		"test",
+		"kotlin",
+		"org",
+		"example"
+	)
+
+	local classes = {
+		["org.example.KotestDescribeSpec"] = "/home/user/project/org/example/KotestDescribeSpec.kt",
+		["org.example.inner.inner2.inner3.KotestDescribeSpec"] = "/home/user/project/org/example/inner/inner2/inner3/KotestDescribeSpec.kt",
+		["org.example.inner.KotestDescribeSpec"] = "/home/user/project/org/example/inner/KotestDescribeSpec.kt",
+	}
+
+	describe("determine_all_classes", function()
+		nio.tests.it("directory", function()
+			local actual = output_parser.determine_all_classes(example_project_path)
+
+			assert.not_nil(actual["org.example.KotestDescribeSpec"])
+			assert.is_true(vim.startswith(actual["org.example.KotestDescribeSpec"], example_project_path))
+		end)
+
+		nio.tests.it("file", function()
+			local test_path = vim.fs.joinpath(example_project_path, "KotestDescribeSpec.kt")
+			local actual = output_parser.determine_all_classes(test_path)
+
+			assert.not_nil(actual["org.example.KotestDescribeSpec"])
+			assert.is_true(vim.startswith(actual["org.example.KotestDescribeSpec"], test_path))
+		end)
+	end)
 
 	describe("parse_lines", function()
-		it("complete example", function()
-			output_parser.parse_lines({
+		nio.tests.it("complete example", function()
+			local actual = output_parser.parse_lines({
```

Comment:
I'm not super familiar with `nvim-nio`, but this took me way too long to find out that you need to use `nio.tests.it` in order to run anything that uses nio; otherwise, your test just runs forever.